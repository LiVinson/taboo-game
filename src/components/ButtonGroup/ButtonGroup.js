import React from "react"
import PropTypes from "prop-types"

//Appears at bottom of Taboo Card when a buttons prop is passed to taboo.
// Receives array of objects with button text and click handler
export default function ButtonGroup({ buttons }) {
  const { handleClick: leftHandleClick, text: leftText } = buttons[0]
  const { handleClick: rightHandleClick, text: rightText } = buttons[1]

  return (
    <div>
      <button onClick={leftHandleClick}>{leftText}</button>
      <button onClick={rightHandleClick}>{rightText}</button>
    </div>
  )
}

ButtonGroup.propTypes = {
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      handleClick: PropTypes.func.IsRequired,
    })
  ),
}
