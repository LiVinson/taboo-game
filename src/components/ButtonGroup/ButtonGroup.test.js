import React from "react"
import {mount} from "enzyme"
import renderer from "react-test-renderer"

import ButtonGroup from "./ButtonGroup"

it("renders correctly", () => {
    const props = {
        buttons: [{text: "Back", handleClick: jest.fn()}, {text: "Next", handleClick: jest.fn()}]
    }
    const ButtonGroupComponent = renderer.create(<ButtonGroup {...props}/>).toJSON();
 
    expect(ButtonGroupComponent).toMatchSnapshot()
})

it("receives a buttons array prop", () => {
    const props = {
        buttons: [{text: "Back", handleClick: jest.fn()}, {text: "Next", handleClick: jest.fn()}]
    }
    const ButtonGroupComponent = mount(<ButtonGroup {...props}/>)

    expect(ButtonGroupComponent.prop("buttons")).toHaveLength(2)
})