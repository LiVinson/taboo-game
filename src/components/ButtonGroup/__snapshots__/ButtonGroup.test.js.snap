// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    onClick={[MockFunction]}
  >
    Back
  </button>
  <button
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;
